uxarray/grid.py

```diff
@@ -660,11 +660,11 @@ def _populate_lonlat_coord(self):
                 "units": "degrees_north",
             })
 
-    def _build_face_dimension(self):
-        """Constructs ``Mesh2_face_dimension``, which calculates the dimension
-        of each face in ``Mesh2_face_nodes``"""
+    def _build_nNodes_per_face(self):
+        """Constructs ``nNodes_per_face``, which contains the number of non-
+        fill-value nodes for each face in ``Mesh2_face_nodes``"""
 
-        # Triangular Mesh
+        # Triangular Mesh (No Fill Values)
```

Comment:
Why the branching condition for triangular mesh is `if not hasattr(self, "nMaxMesh2_face_nodes")?`

In other words, in the future, is that possible that some input files/input parameters don't have the `nMaxMesh2_face_nodes` attribute and also have some fill values?

Reply:
This is related to what we discussed on Tuesday during our meeting. Since there is an instance where we read in a triangular mesh, it will not have a `nMaxMesh2_face_nodes`. I will fix this in a separate request.

Reply:
I understand what you're talking about the MPAS data.
 
 My concern is: this branching is not very strict and robust. If some inputs(not MPAS triangulars) don't have the nMaxMesh2_face_nodes attribute and also have some fill values, it will break this function.
 
 You can also consider add this as a test case for this function using the `Grid.__from_vert__()`

Reply:
I agree. I am planning to remove this branching with a PR that addresses #311

> nMaxMesh2_face_nodes attribute and also have some fill values

I don't believe that this would ever happen since always have `nMaxMesh2_face_nodes` when fill values are present